src/Validation/Validator.php

```diff
@@ -556,9 +573,13 @@ protected function _processRules($field, ValidationSet $rules, $data, $newRecord
                 continue;
             }
 
-            $errors[$name] = $message;
             if (is_string($result)) {
                 $errors[$name] = $result;
+            } elseif ($this->_useI18n) {
+                $args = $rule->get('pass');
+                $errors[$name] = __d($this->_validationDomain, $name, $this->_translateArgs($args));
```

Comment:
Ok, but couldn't you get the same benefits by setting the message to `__d('my_domain', 'email')`?